components/items/ItemsGrid/ItemsGridImage.vue

```diff
@@ -37,6 +43,17 @@ onBeforeMount(async () => {
   const metadata = (await processSingleMetadata(
     props.nft.metadata
   )) as NFTWithMetadata
+
+  if (urlPrefix.value === 'rmrk2' && props.nft.resources.length) {
+    const { src, thumb } = props.nft.resources[0]
+    console.log(src, thumb)
+    item.value = {
+      ...props.nft,
+      image: sanitizeIpfsUrl(thumb || src || ''),
+    }
+    return
```

Comment:
:/

Reply:
40 LOC, IMO we should make it a composable or check it before it come through this component.
And empty return statement leaves me a little confused

Reply:
> And empty return statement leaves me a little confused

Yes please this good thing to introduce some undefined behaviour